analytics/src/analytics/datasets/sprint_board.py

```diff
@@ -63,6 +63,32 @@ def sprint_end(self, sprint: str) -> pd.Timestamp:
         sprint_end = self.df.loc[sprint_mask, self.sprint_end_col].max()
         return sprint_end.tz_localize("UTC")
 
+    @property
+    def sprints(self) -> pd.DataFrame:
+        """Return the unique list of sprints with their start and end dates."""
+        sprint_cols = [self.sprint_col, self.sprint_start_col, self.sprint_end_col]
+        return self.df[sprint_cols].drop_duplicates()
+
+    @property
+    def current_sprint(self) -> str | None:
+        """Return the name of the current sprint, if a sprint is currently active."""
+        return self.get_sprint_name_from_date(pd.Timestamp.today())
+
+    def get_sprint_name_from_date(self, date: pd.Timestamp) -> str | None:
+        """Get the name of a sprint from a given date, if that date falls in a sprint."""
+        # fmt: off
+        date_filter = (
+            (self.sprints[self.sprint_start_col] <= date)  # after sprint start
+            & (self.sprints[self.sprint_end_col] > date)  # before sprint end
+        )
+        # fmt: on
+        matching_sprints = self.sprints.loc[date_filter, self.sprint_col]
+        # if there aren't any sprints return None
+        if len(matching_sprints) == 0:
```

Comment:
if `len > 1`, should we raise an error rather than silently choosing the first one?

Reply:
Oh good point! I'll add a quick fix for this before merging